lib/galaxy/datatypes/text.py

```diff
@@ -856,3 +858,40 @@ def sniff_prefix(self, file_prefix):
                 return False
             found_valid_lines = True
         return found_valid_lines
+
+
+@build_sniff_from_prefix
+class Yaml(Text):
+    """Yaml files"""
+    file_ext = "yaml"
+
+    def sniff_prefix(self, file_prefix):
+        """
+            Try to load the string with the yaml module. If successful it's a yaml file.
+        """
+        return self._looks_like_yaml(file_prefix)
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/yaml'
+
+    def _looks_like_yaml(self, file_prefix):
+        # Pattern used by SequenceSplitLocations
+        if file_prefix.file_size < 50000 and not file_prefix.truncated:
+            # If the file is small enough - don't guess just check.
+            try:
+                item = yaml.safe_load(file_prefix.contents_header)
+                assert isinstance(item, (list, dict))
```

Comment:
Is the assertion necessary, or just  return false

Reply:
Good idea

Reply:
I guess without that, it passes safe_load as a string